// leaflet-timedimension/spec/karma.conf.js
// Karma configuration
module.exports = function (config) {
	var libSources = [
		"src/leaflet.timedimension.js",
		"src/leaflet.timedimension.util.js",
		"src/leaflet.timedimension.layer.js",
		"src/leaflet.timedimension.layer.wms.js",
		"src/leaflet.timedimension.layer.geojson.js",
		"src/leaflet.timedimension.player.js",
		"src/leaflet.timedimension.control.js"
	];
	var leafletSources = require(__dirname+'/../node_modules/leaflet/build/build.js').getFiles(); // Caution Leaflet 1.0.0-beta.2 build needs magic-string

	// base path, that will be used to resolve files and exclude
	basePath = '';
	for (var i=0; i < leafletSources.length; i++) {
		leafletSources[i] = "node_modules/leaflet/" + leafletSources[i];
	}

	depsSources = [
		"node_modules/jquery/dist/jquery.min.js",
		"examples/js/vendors/jquery-ui-1.10.2.min.js",
		"bower_components/iso8601-js-period/iso8601.min.js"
	];

	var files = [
		"spec/sinon.js",
		"spec/expect.js"
	].concat(leafletSources, depsSources, libSources, [
		"spec/after.js",
		"node_modules/happen/happen.js",
		"spec/suites/SpecHelper.js",
		"spec/suites/**/*.js"
	]);

	config.set({
		// base path, that will be used to resolve files and exclude
		basePath: '../',

		plugins: [
			'karma-mocha',
			'karma-coverage',
			'karma-phantomjs-launcher',
			'karma-chrome-launcher',
			'karma-safari-launcher',
			'karma-firefox-launcher'],

		// frameworks to use
		frameworks: ['mocha'],

		// list of files / patterns to load in the browser
		files: files,
		exclude: [],

		// test results reporter to use
		// possible values: 'dots', 'progress', 'junit', 'growl', 'coverage'
		reporters: ['dots'],

		// web server port
		port: 9876,

		// level of logging
		// possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
		logLevel: config.LOG_WARN,

		// enable / disable colors in the output (reporters and logs)
		colors: true,

		// enable / disable watching file and executing tests whenever any file changes
		autoWatch: false,

		// Start these browsers, currently available:
		// - Chrome
		// - ChromeCanary
		// - Firefox
		// - Opera
		// - Safari (only Mac)
		// - PhantomJS
		// - IE (only Windows)
		browsers: ['PhantomJS'],

		// If browser does not capture in given timeout [ms], kill it
		captureTimeout: 5000,

		// Workaround for PhantomJS random DISCONNECTED error
		browserDisconnectTimeout: 10000, // default 2000
		browserDisconnectTolerance: 1, // default 0

		// Continuous Integration mode
		// if true, it capture browsers, run tests and exit
		singleRun: true
	});
};
